src/main/java/com/celements/web/plugin/cmd/PageLayoutCommand.java

```diff
@@ -404,7 +405,7 @@ String getDefaultLayout() {
     return defaultLayout;
   }
 
-  private SpaceReference decideLocalOrCentral(SpaceReference layoutSpaceRef) {
+  public Optional<SpaceReference> resolveValidLayoutSpace(SpaceReference layoutSpaceRef) {
     if ((layoutSpaceRef != null) && !layoutExists(layoutSpaceRef)) {
       SpaceReference centralLayoutSpaceRef = new SpaceReference(layoutSpaceRef.getName(),
```

Comment:
Touching this method and moreover making it public ... can you please refactor this using RefBuilder?